Prometheus.HttpExporter.AspNetCore/HttpRequestMiddlewareBase.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Prometheus.Advanced;
+
+namespace Prometheus.HttpExporter.AspNetCore
+{
+    public class HttpRequestMiddlewareBase<T> where T : ICollector
```

Comment:
It would be nice to see some documentation comments on this class, to outline the basic operating princioples of the label-name-based logic that is implemented. As-is it requires reading the code to discover that the choice of executed logic is driven by labels with specific names.